humility-cmd/src/idol.rs

```diff
@@ -615,55 +618,48 @@ pub fn lookup_reply<'a>(
         .ok_or_else(|| anyhow!("unknown operation \"{}\"", op))?
         .reply;
 
-    let lookup_ok =
-        |ok| {
-            match hubris.lookup_basetype_byname(ok) {
-                Ok(goff) => Ok(*goff),
-                Err(basetype_err) => match m.lookup_enum_byname(hubris, ok) {
-                    Ok(e) => Ok(e.goff),
-                    Err(enum_err) => match m.lookup_struct_byname(hubris, ok) {
-                        Ok(s) => Ok(s.goff),
-                        Err(struct_err) => {
-                            //
-                            // As a last ditch, we look up the REPLY type.
-                            // This is a last effort because it might not be
-                            // there:  if no task calls the function, the type
-                            // will be absent.
-                            //
-                            let t = format!("{}_{}_REPLY", iface.name, op);
-
-                            match hubris.lookup_struct_byname(&t) {
-                                Ok(s) => Ok(s.goff),
-                                Err(reply_err) => {
-                                    //
-                                    // If all of that has failed, we want to
-                                    // generate an error message that contains
-                                    // all of the failures we encountered:
-                                    // this is (much) more likely to be due to
-                                    // Humility limitations rather than
-                                    // anything else.
-                                    //
-                                    bail!(
-                                        "no type for {}.{}: {:?} (as basetype: \
-                                        \"{:?}\"; as enum: \"{:?}\"; as \
-                                        struct: \"{:?}\"; as REPLY: \"{:?}\")",
-                                        iface.name, op, reply, basetype_err,
-                                        enum_err, struct_err, reply_err
-                                    );
-                                }
-                            }
+    let lookup_ok_err = |ok| {
+        if let Err(e) = m.lookup_enum_byname(hubris, ok) {
+            Err(e)
+        } else if let Err(e) = m.lookup_struct_byname(hubris, ok) {
+            Err(e)
+        } else {
+            bail!("no type for {}.{op}", iface.name);
+        }
```

Comment:
Since we just want to bail on errors, could we do this instead?
```suggestion
        m.lookup_enum_byname(hubris, ok)?;
        m.lookup_struct_byname(hubris, ok)?;
        bail!("no type for {}.{op}", iface.name);
```